src/wallet/rpcwallet.cpp

```diff
@@ -294,23 +294,27 @@ static RPCHelpMan getnewaddress()
         label = LabelFromValue(request.params[0]);
 
     OutputType output_type = pwallet->m_default_address_type;
-    bool force_blind = false;
+    // default blinding to the blindedaddresses setting
+    bool add_blinding_key = gArgs.GetBoolArg("-blindedaddresses", g_con_elementsmode);
+
     if (!request.params[1].isNull()) {
         if (!ParseOutputType(request.params[1].get_str(), output_type)) {
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Unknown address type '%s'", request.params[1].get_str()));
         }
         if (output_type == OutputType::BECH32M && pwallet->GetLegacyScriptPubKeyMan()) {
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Legacy wallets cannot provide bech32m addresses");
         }
-        // Special case for "blech32" when `-blindedaddresses=0` in the config.
         if (request.params[1].get_str() == "blech32") {
-            force_blind = true;
+            // always blind for "blech32" even if `-blindedaddresses=0` in the config.
+            add_blinding_key = true;
+        } else if (output_type == OutputType::BECH32) {
```

Comment:
It seems to me that this condition ought to be `request.params[1].get_str() == "bech32"`  to match the branch above.

A careful analysis of the logic has convinced me that what you have written here is logically the same as my proposal. (If `output_type` is `OutputType::BECH32`, then `request.params[1].get_str()` is either `"bech32"` or `"blech32"`, and if we are in this else clause then `request.params[1].get_str() != "blech32"`, and hence `request.params[1].get_str() == "bech32"`.)  But still, it seems unusual to consider `request.params[1].get_str()` in one branch but not the other.